optaplanner-examples/src/main/java/org/optaplanner/examples/common/experimental/impl/ConsecutiveIntervalInfoImpl.java

```diff
@@ -0,0 +1,282 @@
+        ConsecutiveIntervalInfo<Interval_, Point_, Difference_> {
+    private final NavigableMap<IntervalSplitPoint<Interval_, Point_>, IntervalClusterImpl<Interval_, Point_, Difference_>> clusterStartSplitPointToCluster;
+    private final NavigableSet<IntervalSplitPoint<Interval_, Point_>> splitPointSet;
+    private final NavigableMap<IntervalSplitPoint<Interval_, Point_>, IntervalBreakImpl<Interval_, Point_, Difference_>> clusterStartSplitPointToNextBreak;
+    private final Iterable<IntervalCluster<Interval_, Point_, Difference_>> intervalClusterIterable;
+    private final BiFunction<Point_, Point_, Difference_> differenceFunction;
+    private final Iterable<IntervalBreak<Interval_, Point_, Difference_>> breaksIterable;
+
+    public ConsecutiveIntervalInfoImpl(TreeSet<IntervalSplitPoint<Interval_, Point_>> splitPointSet,
+            BiFunction<Point_, Point_, Difference_> differenceFunction) {
+        clusterStartSplitPointToCluster = new TreeMap<>();
+        clusterStartSplitPointToNextBreak = new TreeMap<>();
+        intervalClusterIterable = new MapValuesIterable<>(clusterStartSplitPointToCluster);
+        breaksIterable = new MapValuesIterable<>(clusterStartSplitPointToNextBreak);
+        this.splitPointSet = splitPointSet;
+        this.differenceFunction = differenceFunction;
+    }
+
+    protected void addInterval(Interval<Interval_, Point_> interval) {
+        var intersectedIntervalClusterMap =
+                clusterStartSplitPointToCluster.subMap(
+                        ObjectUtils.defaultIfNull(clusterStartSplitPointToCluster.floorKey(interval.getStartSplitPoint()),
+                                interval.getStartSplitPoint()),
+                        true, interval.getEndSplitPoint(), true);
+
+        // Case: the interval cluster before this interval does not intersect this interval
+        if (!intersectedIntervalClusterMap.isEmpty()
+                && intersectedIntervalClusterMap.firstEntry().getValue().getEndSplitPoint()
+                        .isBefore(interval.getStartSplitPoint())) {
+            // Get the tail map after the first cluster
+            intersectedIntervalClusterMap = intersectedIntervalClusterMap.subMap(intersectedIntervalClusterMap.firstKey(),
+                    false, intersectedIntervalClusterMap.lastKey(), true);
+        }
+
+        if (intersectedIntervalClusterMap.isEmpty()) {
+            // Interval does not intersect anything
+            // Ex:
+            //     -----
+            //----       -----
+            createNewIntervalCluster(interval);
+            return;
+        }
+
+        // Interval intersect at least one cluster
+        // Ex:
+        //      -----------------
+        //  ------  ------  ---   ----
+        var firstIntersectedIntervalCluster =
+                intersectedIntervalClusterMap.firstEntry().getValue();
+        IntervalSplitPoint<Interval_, Point_> oldStart = firstIntersectedIntervalCluster.getStartSplitPoint();
```

Comment:
While `var` has its benefits, I have to say I am not too excited when it's applied inconsistently. Especially if the inconsistent usage is co-located as it is here.

I understand that the compiler sometimes can not infer the type and this explicit type is necessary. In those situations though, I'm not sure if `var` actually brings anything other than inconsistency.

Reply:
I didn't see it; it can be a var